rune/ValueRune.ts

```diff
@@ -151,6 +163,28 @@ class RunMap<T1, U, T2> extends Run<T2, U> {
   }
 }
 
+class RunMatch<T, T2 extends T, T3, U, U2> extends Run<Exclude<T, T2> | T3, U | U2> {
```

Comment:
Yeah, this should be called `RunHandle`, and the existing one should be renamed to `RunRehandle`